### ISSUE TITLE
Towns of Marthas Vineyard: locality versus localadmin

### ISSUE THREAD

**Issue by stepps00** [whosonfirst-data/whosonfirst-data#538]
The `localadmin` features on Marthas Vineyard should be reclassified as `locality` features. We are also missing neighbourhood features (ex: Katama). 

https://whosonfirst.mapzen.com/spelunker/id/102084629/descendants/?exclude=nullisland&placetype=localadmin#11/41.3992/-70.8024

**Comment by nvkelso** [comment]
Let's hold off on this until we have a more general plan. One option being considered is to dup some `localadmin` in the USA to have coterminous `locality` features.

**Comment by nvkelso** [comment]
**tl;dr** 

_Proposal 1a:_ Most `localadmin` in 12 New England and upper-midwest United State `region` can be dup'd to form a coterminous `locality`. The remaining features in those 12 states should remain `localadmin` alone, and the `localadmin` features in the other 6 states should remain as-is. This gets pretty complicated internationally – often there are `locality` tagged inside the `localadmin` that aren't coterminous (see anywhere in Europe), no action proposed outside of USA at this time. 

_Propsoal 1b:_ It is still valid for neighbourhoods to be parented by `county` – there are plenty of unincorporated subdivisions or smaller places that we have data for now in WOF that don't have a `locality` parent but _should_ have a `county` parent during PIPing. For the other 6 states, seems like would be useful but not mission critical for `localadmin` to be included in PIPing. It's more of a concern in Europe.

===

The reason New England "town" are excluded from Quattroshapes derived `locality` features (instead they are in `localadmin`) is we got our data from US Census and they specifically prohibit the treatment as New England "towns" as "incorporated place" [[1](https://www.census.gov/geo/reference/gtc/gtc_place.html)] and neither do they treat them as "census designated places". We can of course do something different in Who's On First, but the history is useful to understand how we got here. And this treatment is consistent with what we do in other countries in Europe, Australia, and well the rest of the world.

To recap: Because US Census treats the New England "town" as a "minor civil division", Quattroshapes passed along to us WOF as `localadmin`. 

**Diving deeper...** Who's On First has `localadmin` for 18 regions in the United States:

<img width="928" alt="screen shot 2016-11-29 at 13 39 02" src="https://cloud.githubusercontent.com/assets/853051/20734096/21fce804-b64c-11e6-82d1-c41fdb72b26f.png">

We want to balance what a normal human would consider a `locality` (probably the "town") with also needing to take US Census updates in the future. 

But not all of those `localadmin` are treated equally _between_ regions, and even _inside_ the same region (e.g. Massachusetts), not all `localadmin` are equal. In the United States some localadmin are more like `locality`, and some more like generic `localadmin`. So let's break that down...

Says Census [[2](http://www.census.gov/geo/reference/gtc/gtc_cousub.html)]:

`Minor civil divisions (MCDs) are the primary governmental or administrative divisions of a county in many states (parishes in Louisiana) and the county equivalents in Puerto Rico and the Island Areas.  MCDs in the United States, Puerto Rico, and the Island Areas represent many different kinds of legal entities with a wide variety of governmental and/or administrative functions.  MCDs include areas variously designated as barrios, barrios-pueblo, boroughs, charter townships, commissioner districts, election districts, election precincts, gores, grants, locations, magisterial districts, parish governing authority districts, plantations, purchases, reservations, supervisor's districts, towns, and townships.  The Census Bureau recognizes MCDs in 29 states, Puerto Rico, and the Island Areas.  The District of Columbia has no primary divisions and is considered equivalent to an MCD for statistical purposes.  (It is also considered a state equivalent and a county equivalent.)`

_We only show 18 (versus 29) on the map above for forgotten reasons._

**Reading on** (and more important to us when discussing coterminous `locality`):

`The MCDs in 12 states (Connecticut, Maine, Massachusetts, Michigan, Minnesota, New Hampshire, New Jersey, New York, Pennsylvania, Rhode Island, Vermont, and Wisconsin) also serve as general-purpose local governments that can perform the same governmental functions as incorporated places.`

Highlighted in purple outline here (green = town, grey = township, white = unincorporated):

<img width="941" alt="screen shot 2016-11-29 at 13 39 15" src="https://cloud.githubusercontent.com/assets/853051/20734123/4df0e6cc-b64c-11e6-9beb-3ed97db90c83.png">

These states do include unincorporated (`UT`) types (which we should **not** map to a coterminous `locality`, and `city` types which are probably already present in WOF and shouldn't be duplicated again (they could be marked coterminous, though).

There are 18 different `qs:types` values which source from US Census (2 of which look to be spelling dups):

<img width="94" alt="screen shot 2016-11-29 at 13 14 30" src="https://cloud.githubusercontent.com/assets/853051/20734135/5594c2c2-b64c-11e6-9cdb-5fd46ed4f6b9.png">

Here blue is type `city`, which we probably already have as `locality`. Others are `town` which could be mapped to a dup'd coterminous `locality`:

<img width="775" alt="screen shot 2016-11-29 at 16 05 35" src="https://cloud.githubusercontent.com/assets/853051/20734414/26656ab8-b64e-11e6-9be7-16c000746b4c.png">

See also:

1. https://www.census.gov/geo/reference/gtc/gtc_place.html
2. http://www.census.gov/geo/reference/gtc/gtc_cousub.html

**Comment by thisisaaronland** [comment]
Things that definitely smell like localities: city, loc, muny, town, village and twp assuming that's short for "township".

Precincts smell like constituencies and there's a larger philosophical question to be have about reservations but if the test is "more like this than that" then I'd agree they should be localadmins. Likewise unincorporated things.

If the goal is to be able to track changes from the census then we should figure out what we need to include in the properties hash to be able to key off of. Bending everything to the Census' worldview doesn't seem like a great idea, particularly if it means that all kinds of things will be missing a locality (specifically an ancestor that is part of the "common" hierarchy).

The census may have perfectly good reasons for believing that the towns on Martha's Vineyard are "minor civil divisions" but that is not how local (or tourists for that matter) understand them.

I am not suggesting that we blanket change all localadmins (even just those in the US) to be localities, only those where it makes sense. I am not sure what making them co-terminous with a cloned localadmin gets us. It doesn't reflect any administrative reality, only the Census Bureau's internal house-keeping. Again, if we need to track changes to Census it seems as though we ought to be able to do that with `fips` concordances or a bespoke property.

**Comment by nvkelso** [comment]
The goal isn't to track changes from US Census per say (I agree that could be done with other means). It's that a ton of US Census data comes aggregated at the existing `localadmin` placetype level and we need to support data visualization as a continuous fabric at that level without forcing downstream data consumers to jump thru a bunch of hoops to get back to to there. 

Back to the placetype hierarchy: `In some states, all or some incorporated places are subordinate to the MCDs in which they are located. Therefore, a place may be either independent of or dependent within MCDs.` [[1](https://www.census.gov/popest/about/geo/terms.html)]. It's entirely possible for there to be _multiple_ valid `locality` features within the existing `localadmin` in the USA, and that is certainly true in our European data. It's also true that `localadmin` can be "unitary" with "town" and should be _also_ made into a `locality` and linked up as "coterminous".

Additionally, the reason US Census treats them as `localadmin` is: `Like civil townships, but unlike most incorporated municipalities in other states, New England towns do not usually represent a single compact populated place.` [[2](https://en.wikipedia.org/wiki/New_England_town#Towns)] which is consistent with Quattroshape's approach in Europe, Australia, and elsewhere.

Since I can't find a goal list for Who's On First geometries, here's my list:

- **search** for place by text (free or structured, e.g. forward geocoding) and getting back a (label) lat-long, (label) bounding box, full geometry, and hints about "hierarchy label" generation. 
- **reverse geocode** of lat-long to a place (preferring the "biggest" shape inclusive of territorial waters)
- **data visualization** based on polygon features (e.g. choropleth, with some sophistication around water and land parts, preferring the land parts)
- **label basemap** "points" (e.g.: `locality` townspots + text) and "polygons" (e.g.: `region` area text). It's important to know which "locality" features `do not usually represent a single compact populated place` so their `mz:min_zoom` can be properly set, and we can choose the correct "point" or "polygon" labeling (and possibly switch between them at different zooms)
- **showing boundary lines on a basemap** (e.g. `region` boundary, `disputed` area boundaries, with some sophistication around water and land parts – e.g. option to not show territorial water parts, coastline parts, and/or inland water parts).

We get more inbound interest in search, reverse geo, and data visualization than the other use cases. I'm looking for a solution that satisfies search & data visualization today, and allows for sophisticated basemap labeling in the future. For instance, for the "same" feature with two representations the `locality` label lat/lng should be on the center of population, while the `localadmin` feature's label lat/lng should be centered in the geographic shape (e.g.: raw MapShaper output).

Here's the distribution of those `type` values, with their (rough geographic distribution):

<img width="314" alt="screen shot 2016-11-29 at 22 30 27" src="https://cloud.githubusercontent.com/assets/853051/20742159/84e0519a-b683-11e6-8989-831cb5b3496e.png">

As we're interesting in `Connecticut, Maine, Massachusetts, Michigan, Minnesota, New Hampshire, New Jersey, New York, Pennsylvania, Rhode Island, Vermont, and Wisconsin` for nomination as localities, that list further breaks down like:

**Nominate for `locality` + `localadmin`** in just a few of the 12 states:

* `pltn` (Maine) - _[extremely tiny](https://en.wikipedia.org/wiki/List_of_plantations_in_Maine) rural localities_
* `Resvn` (N.Y., Maine) - _these are big / important Indian reservations when nothing else is around._
* `town` (New England, N.Y., Wisc.) - _the genesis of this issue_

Some would probably need their default names looked at. For instance, most plantation features in Maine should include ` Planation` in their name. Apple and Google generally don't postfix ` Town` onto the towns in New England (we shouldn't either).

I think label hierarchy building in Pelias already infers ignoring label generation if `coterminous` is set (but there's no harm in setting both properties).

I'm curious to hear @kkowalsky's take on Wisconsin and if "towns" should to be labeled as localities there (always, some of the time, or never).

**Link existing `locality` with existing `localadmin` in the 12 states:**

* `city` (everywhere) - _likely there is already a `locality` in WOF, only the same-as link needs to be made_
* `muny` ([Pa.](https://en.wikipedia.org/wiki/Local_government_in_Pennsylvania#Municipalities)) - a few upgraded boro around Pittsburgh, like [Murrysville](https://en.wikipedia.org/wiki/Murrysville,_Pennsylvania)
* `village` ([Wisc.](https://en.wikipedia.org/wiki/Administrative_divisions_of_Wisconsin#Village), [Mich.](https://en.wikipedia.org/wiki/Administrative_divisions_of_Michigan#Villages), not-N.E.)

**Not sure:**

* `boro` (Penn, N.J.) - many, maybe all, are already `locality` via US Census CDPs in Quattroshapes. _**note** N.Y.C. gets special treatment._
* `chartwp` ([Mich.](https://en.wikipedia.org/wiki/Charter_township)) - while superficially these have the power of `locality`, there are already `locality` features that cover some or parts of some (in the population center) of these places. Apple and Google sometimes label these, and sometimes not, and they aren't consistent in approach.
* `twp` (N.J., [Penn.](https://en.wikipedia.org/wiki/Township_(Pennsylvania)) - some, but a minority, are already `locality` via US Census CDPs in Quattroshapes. _**note** N.H., Mich., and Minn. get different treatment._

Note that Google Maps and Apple Maps do not label townships on their basemap (or even return geometry in Google's case) in Pennsylvania. (e.g.: `West Lampeter Township` where the fair from MapzenCon 2016). Instead the Census Designated Places (like `Willow Street`) are labeled (and we already have those CDPs as `locality` in WOF). 

In both Google Maps and Apple Maps, New Jersey has some but not all townships are labeled (sometimes inclusive of their ` Township` text). They aren't consistent about which ones, or which get ` Township`).

My read is some "1st class" / "home rule" townships in Penn. might be eligible for WOF `locality` status, but many others don't pass the sniff test (would remain `localadmin` only). But that level of granularity isn't present in our WOF data now, we'd have to do a data join with external data. Or we could do a spatial join of `locality` and `localadmin` and for matches where the name is the same then link up only, else create new `locality` for different names, and new `locality` features when no match on spatial join (or intersect).

I'm curious to here from @trescube if he expects to see these Township `localadmin` in Penn. and N.J. labeled at `locality`?

**Ignored** for those 12 states:

* `boro` (N.Y.C.) - we've already manually adjusted this to fit Who's On First's tastes.
* `gore` (Vt.) - These 4 Vermont features are survey book keeping, with extremely low population and large area. There's no where to put a "townspot", ergo it doesn't get a `locality`.
* `grant` (N.H.) - These 9 New Hampshire features are [similar](https://en.wikipedia.org/wiki/Martin%27s_Location,_New_Hampshire) `gore`, but in the neighboring region.
* `loc` (N.H.) - Ditto.
* `purch` (N.H.) - Ditto
* `twp` (N.H., [Mich.](https://en.wikipedia.org/wiki/Administrative_divisions_of_Michigan#Township), Minn.) - Ditto for N.H. In the case of Michigan only the "charter township" are locality worthy. For Minnesota there are already "city" features in most of the township areas that are already `locality` for us (e.g. [Fisher_Township](https://en.wikipedia.org/wiki/Fisher_Township,_Polk_County,_Minnesota) versus Fisher the town in the same county).
* `UT` (Maine, Minn.) - This is US Census book keeping only. (same as UT below)

I'm curious to hear from @meghanhade and @rfriberg if they expect Minnesota townships to be labeled as localities. 

**Not eligible** for `locality` because they aren't in the 12 states where `localadmin` are generally treated as functioning like a "city" or "town":

* `Dist` (Nebr.)
* `prec` (Nebr., Ill.)
* `Prec` (Nebr., Ill.)
* `Twp` (Kan., Nebr.)
* `UT` (N.D., S.D.) - This is US Census book keeping only.

For instance: Those Nebraska precincts _are_ weird! Looks like US Census does source them from election `constituency` for counties without other subdivisions! Definitely `localadmin` only!

`There are 1,198 county subdivisions in Nebraska that are known as minor civil divisions (MCDs). There are 27 counties with townships, 65 counties with election precincts, and one county with districts. There are 435 active functioning townships and 30 inactive townships that have the ability to activate and perform governmental functions. Each township is governed by a board of thirteen elected officials. There are 646 precincts which are nonfunctioning, nongovernmental county subdivisions used for conducting elections. Webster County has five nonfunctioning, nongovernmental districts used in conducting elections. There are 77 incorporated places that are independent of MCDs and constitute 79 county subdivisions.` [[3](https://www2.census.gov/geo/pdfs/reference/guidestloc/ne_gslcg.pdf)] 

**See also:**

1. https://www.census.gov/popest/about/geo/terms.html
2. https://en.wikipedia.org/wiki/New_England_town#Towns
3. https://www2.census.gov/geo/pdfs/reference/guidestloc/ne_gslcg.pdf

**Comment by nvkelso** [comment]
Talking to Nat he says:

`New England: Towns are admin, definitely. Several county governments in Massachusetts have actually been dissolved (I have a couple friends who are town and (former) county administration: the county still exists, but has no administrative functions. Towns are really the most important sub-state admin unit. They often contain villages or urban centers with specific names, often multiple per town. People think of the town for admin, but will say they live in [village name]. I lived in the town of Thetford, and would say I lived in Thetford or in Thetford Hill. But I worked in White River Junction, which is in the town of Hartford, and would never say I worked in Hartford (there is a village Hartford Four Corners, but I would have said I worked there and not in Hartford). Villages deserve townspots. Pretty much every town will have at least one, where the town hall and post office at least is.`

`Minnesota townships and Wisconsin towns are unincorporated sub-county admin units. They usually have a town hall and some admin functions (but many more functions are carried out by the county). I would generally expect an area label. But they have definite boundaries—they are based on the classic GLO survey 6x6 mile townships. They often contained unincorporated places (which may or may not have the same name). Those should get place spots.`

When I preview the remaining Quattroshapes **point** locality features to import into Who's On First on the island this seems to match up with Nat's description (modulo temporary duplicate point features for Edgartown).

<img width="912" alt="screen shot 2016-11-30 at 09 41 05" src="https://cloud.githubusercontent.com/assets/853051/20764144/b652e716-b6e1-11e6-8e22-6cbd9d967ac0.png">

But looks like there are a lot of ADMD type features from GeoNames.org in that dataset that may need to be sorted thru using the overall guidelines forming up in this issue:

Looking around Madison, Wisc.:

<img width="942" alt="screen shot 2016-11-30 at 09 47 21" src="https://cloud.githubusercontent.com/assets/853051/20765626/f85be0d0-b6e7-11e6-9f05-93a8781b8547.png">

South of Minneapolis-St. Paul:

<img width="939" alt="screen shot 2016-11-30 at 09 48 22" src="https://cloud.githubusercontent.com/assets/853051/20765625/f8448232-b6e7-11e6-859e-8f345d086db8.png">

And near Lancaster, Penn.:

<img width="896" alt="screen shot 2016-11-30 at 10 31 25" src="https://cloud.githubusercontent.com/assets/853051/20765664/2fe1f4fe-b6e8-11e6-8a29-bf48b192646f.png">

**Comment by thisisaaronland** [comment]
I think there are a couple over-lapping issues here:

1. "It's that a ton of US Census data comes aggregated at the existing localadmin placetype level..." - That's just the history of the thing (QS) that WOF inherited from. My personal feeling is that, in WOF terms, all of these places are more like localities than localadmins. That being said in absolute terms it doesn't necessarily make sense or warrant the time and effort to clean up / change everything. In that sense I am fine with making thing co-terminous.

2. The notion of "continuous fabric" - I would argue this applies to the common hierarchy as much as anything else. The whole rationale behind that hierarchy was a reasonable middle ground that would accommodate a global dataset, and not get in the hair-splitting of things like Geonames.

Not being able to find things that people know and experience to be localities (for example Edgartown) because they are a (necessarily) ambiguous placetype called a "localadmin" isn't great. Having to look for "towns" in both localities and localadmins is also not great. It's not that localadmins don't exists it's that some of the things we've labeled localadmins don't make any sense outside of expert usage/experience.

Again, as a short-term solution I am fine with making things co-terminous. Longer term I am not sure that the QS model (which was fine and good for QS) maps well to WOF. If the issue is search and discovery there are lots of ways to address that but I don't know that overloading "localadmin" to mean two different things – specifically for things that are "localities" (and not cities or towns or hamlets) is a good idea.

_As an aside you posted Nat's comments as I am writing this: The QS dots on the Vineyard are pretty weird in that things labeled "town" are in fact towns, a couple of the dots are common parlance for parts of the island and some of them I've never even heard of._

**Comment by nvkelso** [comment]
**Notes from meeting between @thisisaaronland and @nvkelso yesterday:**

- **LocalAdmin**
    - When 100% the same as locality, with the same name
        1. Deprecate the localadmin into the locality
        2. Add "other placetypes" property with array that also says ['localadmin']
            This will get referenced with the metafiles are created (so this feature will appear in multiple metafiles) - on Aaron
        3. We can deal with different label points later (area versus point)
        4. Examples: Amsterdam (this will be test location), Rotterdam, Marthas Vineyard localadmin except Oak Bluffs.
    - When 100% geom same as locality, but different names
        1. Keep 2 records, adding a "co-terminuous" linkage
        2. Example: Washington, DC
    - When < 100% the same
        - Keep as separate records
        - #538 Oak Bluffs              
- **County**
    - San Francisco county deprecated into locality record
    - Adds "other placetypes" : ['county']

I need to flesh out the examples more, and propose some PRs based on it so we have data samples.

**Comment by thisisaaronland** [comment]
https://github.com/whosonfirst-data/whosonfirst-data/compare/14cbbe635cfe...6d4e3b585a9f

**Comment by thisisaaronland** [comment]
https://github.com/whosonfirst/whosonfirst-properties/commit/c1efda9a2450b077ab73673df5cc00ee262d3608

**Comment by nvkelso** [comment]
We should make these places with the new `wof:statistical_gore` = **1** and keep their placetype as-is.

- `gore` (Vt.) - These 4 Vermont features are survey book keeping, with extremely low population and large area. There's no where to put a "townspot", ergo it doesn't get a locality.
- `grant` (N.H.) - These 9 New Hampshire features are similar gore, but in the neighboring region.
- `loc` (N.H.) - Ditto.
- `purch` (N.H.) - Ditto
- `twp` (N.H., Mich., Minn.) - Ditto for N.H. In the case of Michigan only the "charter township" are locality worthy. For Minnesota there are already "city" features in most of the township areas that are already locality for us (e.g. Fisher_Township versus Fisher the town in the same county).
- `UT` (Maine, Minn.) - This is US Census book keeping only. (same as UT below)
- `Dist` (Nebr.)
- `prec` (Nebr., Ill.)
- `Prec` (Nebr., Ill.)
- `Twp` (Kan., Nebr.)
- `UT` (N.D., S.D.) - This is US Census book keeping only.

**Comment by nvkelso** [comment]
Tackling part of this via https://github.com/whosonfirst-data/whosonfirst-data/issues/1586.

**Comment by nvkelso** [comment]
See also in New Jersey: `Weehawken Township` versus `Weehawken`.
